asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -56,45 +77,57 @@ def build_dataset(
         all_fns = glob(f"{in_files}/*complex.pdb")
     else:
         all_fns = glob(in_files)
-    # Extract crystal structure and compound id from file name
-    xtal_pat = r"Mpro-.*?_[0-9][A-Z]"
-    compound_pat = r"[A-Z]{3}-[A-Z]{3}-[0-9a-z]+-[0-9]+"
-
-    xtal_matches = [re.search(xtal_pat, fn) for fn in all_fns]
-    compound_matches = [re.search(compound_pat, fn) for fn in all_fns]
-    idx = [bool(m1 and m2) for m1, m2 in zip(xtal_matches, compound_matches)]
-    compounds = [
-        (xtal_m.group(), compound_m.group())
-        for xtal_m, compound_m, both_m in zip(xtal_matches, compound_matches, idx)
-        if both_m
-    ]
-    num_found = len(compounds)
-    # Dictionary mapping from compound_id to Mpro dataset(s)
-    compound_id_dict = {}
-    for xtal_structure, compound_id in compounds:
-        try:
-            compound_id_dict[compound_id].append(xtal_structure)
-        except KeyError:
-            compound_id_dict[compound_id] = [xtal_structure]
+
+    # Load the experimental compounds
+    exp_data, exp_compounds = load_exp_data(
+        exp_fn,
+        achiral=achiral,
+        return_compounds=True,
+        check_range_nan=check_range_nan,
+        check_stderr_nan=check_stderr_nan,
+    )
+
+    # Parse structure filenames
+    if (model_type.lower() != "2d") or str_only:
```

Comment:
I have been moving some instances of `2d` to be `GAT` does this change make sense/ are you happy with it? If so we could make that change or I can revert some of my changes.

Reply:
yeah I think this makes sense. I've switched everything (I think) over in [e37f0f2](https://github.com/choderalab/asapdiscovery/pull/198/commits/e37f0f2d5607ac2a11bf63f6ec0c01a9c10abd55)